Tests/AK/TestStringFloatingPointConversions.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2022, Dan Klishch <danilklishch@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/BitCast.h>
+#include <AK/StringFloatingPointConversions.h>
+#include <LibTest/TestCase.h>
+
+static bool operator!=(AK::FloatingPointExponentialForm a, AK::FloatingPointExponentialForm b)
+{
+    return a.sign != b.sign || a.exponent != b.exponent || a.fraction != b.fraction;
+}
+
+template<>
+struct AK::Formatter<AK::FloatingPointExponentialForm> : Formatter<FormatString> {
+    ErrorOr<void> format(FormatBuilder& builder, AK::FloatingPointExponentialForm value)
+    {
+        return Formatter<FormatString>::format(builder, "(s={} f={} e={})"sv, value.sign, value.fraction, value.exponent);
+    }
+};
+
+#define DOES_CONVERT_DOUBLE_TO(value, sign, fraction, exponent)                             \
+    do {                                                                                    \
+        EXPECT_EQ(                                                                          \
+            convert_floating_point_to_decimal_exponential_form(static_cast<double>(value)), \
+            (AK::FloatingPointExponentialForm { sign, fraction, exponent }));               \
+    } while (false)
+
+// Tests here only check basic cases. While writing, I mostly relied on the benchmarks and
+// stress tests, which can be found at https://github.com/DanShaders/serenity-arithmetic-benchmark/blob/master/StringFloatingPointConversions.cpp .
+
+TEST_CASE(double_conversion)
+{
+    DOES_CONVERT_DOUBLE_TO(0, 0, 0, 0);
```

Comment:
I guess this will be excercised in the rest of the system but I would still like a couple of more trivial tests like: `0.1` `0.2` `0.3` as those as "famous" examples with `0.1 + 0.2 !== 0.3` in javascript.

Also at least the negative zero edge case.